src/resolution/composite-arns-resolver.ts

```diff
@@ -67,6 +71,82 @@ export class CompositeArNSResolver implements NameResolver {
       registryCache,
       networkProcess,
     });
+    this.limit = pLimit(maxConcurrentResolutions);
+  }
+
+  private async resolveWithResolver(
+    resolver: NameResolver,
+    name: string,
+    baseArNSRecordFn: () => Promise<any>,
+    isLastResolver: boolean,
+  ): Promise<NameResolution | undefined> {
+    try {
+      const resolutionPromise = resolver
+        .resolve({
+          name,
+          baseArNSRecordFn,
+        })
+        .then((resolution) => {
+          if (resolution.resolvedAt !== undefined) {
+            this.resolutionCache.set(
+              name,
+              Buffer.from(JSON.stringify(resolution)),
+            );
+            this.log.info('Resolved name', { name, resolution });
+            return resolution;
+          }
+          return undefined;
+        });
+
+      if (isLastResolver) {
+        return await resolutionPromise;
+      }
+
+      return await pTimeout(resolutionPromise, {
+        milliseconds: config.ARNS_COMPOSITE_RESOLVER_TIMEOUT_MS,
```

Comment:
nit: make this injectible in the constructor, and default to the arns composite resolver timeout. should help validate tests with timeouts

Reply:
Great suggestion! [`2455ae2` (#365)](https://github.com/ar-io/ar-io-node/pull/365/commits/2455ae2fcf8f1326d5b0755123db966368b5e4bf)